commons/src/main/java/fr/free/nrw/commons/upload/MwVolleyApi.java

```diff
@@ -18,23 +19,75 @@
 
 
 import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
 
 public class MwVolleyApi {
 
     private static RequestQueue REQUEST_QUEUE;
     private static final Gson GSON = new GsonBuilder().create();
     private Context context;
+    private static String coordsLog;
+
+    protected static HashSet<String> categorySet;
+
+    //To check later on whether any nearby categories were found
+    public static boolean gpsCatExists;
 
     public MwVolleyApi(Context context) {
         this.context = context;
+        categorySet = new HashSet<String>();
+    }
+
+    //To get the list of categories for display
+    public static ArrayList<String> getGpsCat() {
+        ArrayList<String> list = new ArrayList<String>(categorySet);
+        return list;
     }
-    public void request(String apiUrl) {
+
+
+    public void request(String coords) {
+
+        coordsLog = coords;
+        String apiUrl = buildUrl(coords);
+        Log.d("Image", "URL: " + apiUrl);
+
         JsonRequest<QueryResponse> request = new QueryRequest(apiUrl,
                 new LogResponseListener<QueryResponse>(), new LogResponseErrorListener());
         getQueue().add(request);
     }
 
-    private RequestQueue getQueue() {
+    /**
+     * Builds URL with image coords for MediaWiki API calls
+     * Example URL: https://commons.wikimedia.org/w/api.php?action=query&prop=categories|coordinates|pageprops&format=json&clshow=!hidden&coprop=type|name|dim|country|region|globe&codistancefrompoint=38.11386944444445|13.356263888888888&
+     * generator=geosearch&redirects=&ggscoord=38.11386944444445|13.356263888888888&ggsradius=100&ggslimit=10&ggsnamespace=6&ggsprop=type|name|dim|country|region|globe&ggsprimary=all&formatversion=2
+     */
+    private String buildUrl (String coords){
+
+        Uri.Builder builder = Uri.parse("https://commons.wikimedia.org/").buildUpon();
```

Comment:
The base URL here might be worth extracting as a String constant to encourage reuse when adding new APIs.